src/fvm_lowered_cell_impl.hpp

```diff
@@ -56,6 +56,10 @@ class fvm_lowered_cell_impl: public fvm_lowered_cell {
 
     value_type time() const override { return tmin_; }
 
+    std::vector<mechanism_ptr>& mechanisms() {
```

Comment:
Does this have to be a non-const reference, i.e. can we make the return type `const std::vector<mechanism_ptr>&`?

A con-const reference would allow a users to modify the contents of the class. This isn't necessarily a massive problem, because this is part of our internal API, but it still good to avoid.

Reply:
I see that it does have to be for the benchmark. Maybe we should put a comment above this line that points out that these are exposed for testing purposes only.